checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -1113,34 +1173,71 @@ private void updateObligationsForAssignment(
           && !ElementUtils.isFinal(lhsElement)) {
         checkReassignmentToField(obligations, assignmentNode);
       }
+
       // Remove Obligations from local variables, now that the owning field is responsible.
       // (When obligation creation is turned off, non-final fields cannot take ownership.)
       if (isOwningField
           && rhs instanceof LocalVariableNode
           && (typeFactory.canCreateObligations() || ElementUtils.isFinal(lhsElement))) {
+
+        LocalVariableNode rhsVar = (LocalVariableNode) rhs;
+
+        boolean inConstructor =
+            TreePathUtil.inConstructor(typeFactory.getPath(assignmentNode.getTree()));
```

Comment:
I would be a bit surprised is there isn't a more efficient way to do this.  If needed, I would be in favor of storing the `cfg` passed to `analyze()` in an instance field so we can check this type of condition more efficiently.  See also #5985

Reply:
Is [3dff5b6](https://github.com/typetools/checker-framework/pull/6241/commits/3dff5b69fd2d159c6c3394dde4d5d6f07ecb7038) what you had in mind?

Reply:
Yes, this exactly, thanks!